developer/bake_config.py

```diff
@@ -63,6 +62,8 @@ def _should_skip_caching_sparse(desc):
     :param dict desc: Descriptor to check.
     :returns: ``True`` if the contents should be skipped, ``False`` otherwise.
     """
+    if desc["type"] in ["dev", "path"]:
+        logger.warning("'%s' will not be cached inside the configuration.", desc)
```

Comment:
Why are we not including `"app_store"` here?
Since the return value counts `"app_store"`

Reply:
I had not refactored the code enough. The dev and path descriptors are now skipped from the main caching loop and not by these functors.